### DISCUSSION TITLE
`clear -x` works differently than other terminal emulators

### DISCUSSION THREAD

**Discussion by elyalon** [wezterm/wezterm#5558] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

From `man clear`:

>-x       prevents clear from attempting to clear the scrollback buffer.

I use `clear -x` instead of `clear` because I like keeping the scrollback, and I noticed that all the text that is on screen gets "chopped off" by wezterm, losing that data from the buffer. [video](https://github.com/wez/wezterm/assets/105967627/e21cfc58-54fe-4dd1-bb0c-2c39ae0d5829)

I don't know if this is a bug or not but all other terminals i've tried don't have this "chopping off" thing

**Comment by wez** [comment, marked as answer]
likely the same as https://github.com/wez/wezterm/discussions/4446

**Comment by elyalon** [comment]
Got it, I made this fish function that gives me the behaviour I want

```fish
function clearx
    set -l row (cursor_pos | jq .row)
    and printf "\e[""$row""S"
    and clear -x
end
```

(where `cursor_pos` is this script: https://github.com/ourarash/cursor-position/blob/main/cursor-position.sh)
